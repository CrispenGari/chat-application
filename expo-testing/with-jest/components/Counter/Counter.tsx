import { View, Text, TextInput, Button } from "react-native";
import React from "react";

const Counter = ({ value }: { value: number }) => {
  const [count, setCount] = React.useState(value);
  const [amount, setAmount] = React.useState(0);
  return (
    <View>
      <Text testID="count">{count}</Text>
      <TextInput
        value={amount.toString()}
        onChangeText={(text) => setAmount(parseInt(text))}
        placeholder="Amount"
      />
      <Button title="Increment" onPress={() => setCount((s) => s + amount)} />
      <Button title="Decrement" onPress={() => setCount((s) => s - amount)} />
    </View>
  );
};

export default Counter;
