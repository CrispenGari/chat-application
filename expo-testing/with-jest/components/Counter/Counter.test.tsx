import { render, screen, fireEvent } from "@testing-library/react-native";
import Counter from "./Counter";

describe("<Counter/>", () => {
  it("renders correctly with initial value of 10", () => {
    render(<Counter value={10} />);
    const text = screen.getByText("10");
    const incrementBtn = screen.getByRole("button", { name: /increment/i });
    const decrementBtn = screen.getByRole("button", { name: /decrement/i });
    const textBox = screen.getByPlaceholderText(/amount/i);
    expect(incrementBtn).toBeOnTheScreen();
    expect(textBox).toBeOnTheScreen();
    expect(decrementBtn).toBeOnTheScreen();
    expect(text).toBeOnTheScreen();

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("increments by 5", async () => {
    render(<Counter value={10} />);
    const incrementBtn = screen.getByRole("button", {
      name: /increment/i,
    });
    const textBox = screen.getByPlaceholderText(/amount/i);
    fireEvent.changeText(textBox, "5");
    fireEvent.press(incrementBtn);
    const text = await screen.findByTestId("count");
    expect(text).toHaveTextContent("15");
  });
  it("decrement by 5", () => {
    render(<Counter value={10} />);
    const btn = screen.getByRole("button", {
      name: /decrement/i,
    });
    const textBox = screen.getByPlaceholderText(/amount/i);
    fireEvent.changeText(textBox, "5");
    fireEvent.press(btn);
    const text = screen.getByTestId("count");
    expect(text).toHaveTextContent("5");
  });
});
