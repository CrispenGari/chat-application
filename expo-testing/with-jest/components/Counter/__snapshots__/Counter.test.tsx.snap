// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly with initial value of 10 1`] = `
<View>
  <Text
    testID="count"
  >
    10
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Amount"
    value="0"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement
      </Text>
    </View>
  </View>
</View>
`;
